doc/compiler/optimizer/inlineFastPathLocations.md

```diff
@@ -0,0 +1,119 @@
+Secondary Licenses when the conditions for such availability set
+forth in the Eclipse Public License, v. 2.0 are satisfied: GNU
+General Public License, version 2 with the GNU Classpath
+Exception [1] and GNU General Public License, version 2 with the
+OpenJDK Assembly Exception [2].
+
+[1] https://www.gnu.org/software/classpath/license.html
+[2] http://openjdk.java.net/legal/assembly-exception.html
+
+SPDX-License-Identifier: EPL-2.0 OR Apache-2.0 OR GPL-2.0 WITH Classpath-exception-2.0 OR LicenseRef-GPL-2.0 WITH Assembly-exception
+-->
+
+# Different Inline Fast Path Optimization Locations
+
+
+## Inline Fast Path Optimizations
+
+When there is a call to a method, the call can be executed in a cheaper way rather than making a call out to the target
+method. The inlined fast path converts a recognized call into a cheaper sequence of IL trees. One example is a call to a
+`java.lang.Math.abs()`. It can be converted into an `iabs` (absolute value of integer) [opcode](https://github.com/eclipse/omr/blob/e3a15a993c8aba80582aa1d6f3071e122acbd4c4/compiler/il/OMROpcodes.enum#L1911) .
+
+The call to `java.lang.Math.abs()` can be eliminated because there is a better representation of the execution in IL which
+is cheaper. That is a basic pattern of taking a call, recognizing them in a certain way, and optimizing it, sometimes using
+opcode, sometimes using a sequence of IL trees etc.
+
+
+There are multiple locations in our code where we do inline fast path optimizations:
+
+1. [Inliner](https://github.com/eclipse/omr/blob/master/compiler/optimizer/Inliner.cpp)
+2. [RecognizedCallTransformer](https://github.com/eclipse/omr/blob/master/compiler/optimizer/OMRRecognizedCallTransformer.cpp)
+3. [IL Generator](https://github.com/eclipse-openj9/openj9/tree/master/runtime/compiler/ilgen)
+4. [UnsafeFastPath](https://github.com/eclipse-openj9/openj9/blob/master/runtime/compiler/optimizer/UnsafeFastPath.cpp)
+
+## RecognizedCallTransformer
+
+`RecognizedCallTransformer` is similar to what the Inliner does with respect to how it recognizes calls and converts them
+into sequences of IL trees. However. Inliner is more complicated. There are a lot of things going on with respect to heuristics.
+Inliner constantly checks: Do I have a budget and can I inline this call? There may be checks that prevent us from inlining
+a particular method into another one because too many methods have been inlined already.
+
+`RecognizedCallTransformer` does not need to worry about budget. It is much simpler. It does not have any particular heuristics.
+It performs transformations unconditionally because 99% of the chance it is beneficial to do so.
+
+`RecognizedCallTransformer` is run as part of `ilgenStrategyOpts`. [ilgenStrategyOpts](https://github.com/eclipse/omr/blob/e3a15a993c8aba80582aa1d6f3071e122acbd4c4/compiler/optimizer/OMROptimizer.cpp#L535-L544)
+is run as soon as IL is generated. It also runs on the generated IL for the callee that will be inlined before the callee
```

Comment:
"It also runs on the generated IL for the callee that will be inlined before the callee" does not quite sound right

Reply:
It's rephrased as 

> ilgenStrategyOpts always runs as soon as IL is generated. For example for normal inlining, after ILGen runs to produce the IL for the callee method, ilgenStrategyOpts also runs on the generated IL before the method is inlined into the caller.